inst/include/RcppArmadilloWrap.h

```diff
@@ -149,7 +149,11 @@ namespace Rcpp{
     template <typename T>
     SEXP wrap( const arma::field<T>& data){
         RObject x = wrap( RcppArmadillo::FieldImporter<T>( data ) ) ;
-        x.attr("dim" ) = Dimension( data.n_rows, data.n_cols ) ;
+        # if defined(RCPP_ARMADILLO_FIX_FieldImporter)
+            x.attr("dim" ) = Dimension( data.n_rows, data.n_cols, data.n_slices ) ;
```

Comment:
Shouldn't this be further conditioned based on the dimension attributes of `data` similar to the as call? , e.g. 

```
                if (data.n_elem == 1)
                {
                   x.attr("dim" ) = Dimension(data.n_rows);
                }
                else if (data.n_elem == 2)
                {
                     x.attr("dim" ) = Dimension( data.n_rows, data.n_cols);
                }
                else if (data.n_elem == 3)
                {
                     x.attr("dim" ) = Dimension( data.n_rows, data.n_cols, data.n_slices);
                }
```

If retained as-is, this would force all fields to be an array with 3 dimensions in _R_.

Reply:
Nice catch, thanks for that.